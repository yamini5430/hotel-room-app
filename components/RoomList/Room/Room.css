.room {
    min-height: 100px;
    max-width: 220px;
    border: 3px solid #D3D3D3;
    border-radius: 6px;
    background-color: #D3D3D3;
    padding: 3px 3px;
}
.title {
    margin-bottom: 5px;
}

.room.disabled {
    border : 3px solid #C0C0C0;
    background-color: #DCDCDC;
}
