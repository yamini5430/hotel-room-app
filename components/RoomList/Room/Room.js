import React,{Component} from 'react';
import BuildControl from "../BuildControl/BuildControl";
import Checkbox from "../../UI/CheckBox/CheckBox"

const Room = (props) => {
  return  <div className={"room" + (!props.data.enabled ? " disabled" : "")}>
	   {props.type != "room1" ? <Checkbox label = {props.data.name} type = {props.type} 
	   checkChangeHandler = {props.checkChangeHandler} isChecked = {props.data.enabled} /> :
	   <div className="title">{props.data.name}</div> }
		<BuildControl disabledInfo = {!props.data.enabled} peopleData = {props.data.data}
		updatePersonHandler = {props.updatePersonHandler} type = {props.type} />
		
	</div>
}

export default Room;