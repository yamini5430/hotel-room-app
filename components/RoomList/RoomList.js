import React from 'react';
import Room from './Room/Room';

const RoomList = (props) => {
	const transformedData = Object.keys(props.data).map( igKey => {
		return [...Array(props.data[igKey])].map((data,index) => {
		return <Room type={igKey} key = {data['key']} data = {data} checkChangeHandler = {props.checkChangeHandler}
		updatePersonHandler = {props.updatePersonHandler}/>
	})
	}).reduce( (arr,curr) => arr.concat(curr), []);
  return (
  		<div className = "roomList">
  		{transformedData}
  		</div>
  	)
}

export default RoomList;