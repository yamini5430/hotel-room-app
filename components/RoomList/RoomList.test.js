import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RoomList from "./RoomList";
import Room from "./Room/Room";

configure({adapter : new Adapter})

const wrapper = shallow(<RoomList />);

// Snapshot for RoomList React Component
describe('>>>Room List--- Snapshot',()=>{
  it('+++capturing Snapshot of Room List', () => {
      const wrapper = shallow(<RoomList />);
      expect(wrapper).toMatchSnapshot();
    })
});

describe('>>>Room List--- Elements Getting Rendered Correctly',()=>{
 it('+++renders the room items correctly', () => {
    //expect(wrapper.find(Room)).to.have.length(4);
    expect(wrapper.children().find(Room).length == 4);
    })
});