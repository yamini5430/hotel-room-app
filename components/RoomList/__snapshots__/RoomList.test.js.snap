// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Room List--- Snapshot +++capturing Snapshot of Room List 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoomList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "roomList",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "roomList",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
